v2/internal/progress/stdcapture.go

```diff
@@ -0,0 +1,95 @@
+package progress
+
+/**
+  Inspired by the https://github.com/PumpkinSeed/cage module
+ */
+import (
+	"bytes"
+	"io"
+	"os"
+	"sync"
+)
+
+type captureData struct {
+	backupStdout *os.File
+	writerStdout *os.File
+	backupStderr *os.File
+	writerStderr *os.File
+
+	DataStdOut *bytes.Buffer
+	DataStdErr *bytes.Buffer
+
+	outStdout  chan []byte
+	outStderr  chan []byte
+}
+
+func startStdCapture() *captureData {
+	rStdout, wStdout, errStdout := os.Pipe()
+	if errStdout != nil {
+		panic(errStdout)
+	}
+
+	rStderr, wStderr, errStderr := os.Pipe()
+	if errStderr != nil {
+		panic(errStderr)
+	}
```

Comment:
As seen in `pprof` the `os.Pipe()` call is quite a costly one, so CPU usage can spike if many captures are performed.